### DISCUSSION TITLE
Set the defaults object in client

### DISCUSSION THREAD

**Discussion by sbc640964** [tighten/ziggy#621] [Q&A]
#### Description

Add an option to update the selector object from the client.

#### Suggestion

Add an option to update the selector object from the client.

The defaults object is currently updated on the server side, working with inertiajs for example will not take into account that the object is initialized only for the first time.

I have an application that when I'm inside one of its resources I want the parameter to be default, because it forces me to use it in a lot of places.

If I browse directly to the item resourse page, the default is indeed set, but if I browsed to the index of the resource, the resource I enter will not be the default, since it has already been initialized upon entering the index (as null).

I would appreciate a solution.

#### Alternatives

I don't know if this is correct, but I ended up doing it like this:
```
Ziggy.defaults = {...Ziggy.defaults, [key]: value}
```

**Comment by bakerkretzmar** [comment]
Take a look at https://github.com/tighten/ziggy/discussions/463#discussioncomment-1381011 and read through the discussion in https://github.com/tighten/ziggy/discussions/465 for some approaches to making this work.

Ziggy can't automatically update `defaults` for you without a page reload, because it doesn't know anything about your navigation or the state of your app. The example you shared is one way to handle this, although there are some ideas in the discussions above for cleaner / more scalable ways to do it across your whole app.

Let me know if you have any other questions!
